### ISSUE TITLE
GoodDapp MicroBridge Implementation

### ISSUE THREAD

**Issue by decentralauren** [GoodDollar/GoodProtocolUI#558]
## Business Description
Implement new MicroBridge SDK in GoodDapp. This ticket replaces [this prior issue.](https://github.com/GoodDollar/GoodProtocolUI/issues/252)

## Detailed Description of Functionality
Please see FIgma file (referenced below) for details.

[FAQ Content can be found here](https://www.notion.so/gooddollar/9a11ade7a2e247d3aeee137d7cf07cb8?v=8ce41a466d18486bbfdfa127eea21ded&p=9174071a2d1441449be36ada5fee6170&pm=s) (WIP)

## Technical Implementation
See https://github.com/GoodDollar/GoodWeb3-Mono/issues/157

Gooddapp implementation is just updating the sdk packages.

Specificly needed to be done in dapp:
- [x] - add FAQ's

## Design Reference
[FIgma](https://www.figma.com/design/ihw1PxBvLxacTHnN2aj4lC/3.-Product?node-id=17767-48114)
<img width="1740" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeFl3QlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--dff37757d24eec007ec59e3731a915584a814a49/Screenshot%202024-06-03%20at%2010.48.25%20AM.png" alt="Screenshot 2024-06-03 at 10.48.25 AM.png" />

## Acceptance Criteria
See Epic

## Test Cases
*Specify the happy path and unhappy path scenarios to test to verify the feature works as expected.*

**Comment by decentralauren** [comment]
@L03TJ3 should this also be in sprint (Implementation of MicroBridge into GoodDapp) or think it will need to wait to next sprint?

**Comment by L03TJ3** [comment]
@decentralauren I assume today it could be possible to be done this sprint, but lets keep it in product-backlog for the time being (I'll update in two days if aything changes)

**Comment by vldkhh** [comment]
@L03TJ3 The GoodDapp is crashed after entering 1000G$ amount. It seems like a bug in the calculations .

https://github.com/user-attachments/assets/722c8f05-c447-45d1-a519-54e0edb61384
![Snag_15f527](https://github.com/user-attachments/assets/96264106-37e1-40f3-b42b-f8989f00c3b1)

**Comment by L03TJ3** [comment]
@sirpy Current status of this (and why its moved back to in-progress)
All works, except for rpc's getting blocked by:
1. to many requests
2. block-range too-wide (its now at 20k blocks)


These blocked rpc's break the flow, especially around the recent-transaction list.
I don't know exactly how to handle this better, maybe we could use a cycle/fallback flow as we do in the wallet.

**Comment by sirpy** [comment]
@L03TJ3 
1. we should stick to forno.celo.org and rpc.fuse.io
2. you can reduce block range.
3. too many requests, when does that happen? could be we need to throttle in cases like when user types the amount, we should wait until he finished typing.

**Comment by L03TJ3** [comment]
@sirpy it mainly gives issues around loading the widget / recent claims widget. once its loaded I have not encountered any following rpc call issues

**Comment by sirpy** [comment]
@L03TJ3 I dont understand, is the issue with the bridge or with recent claims?

**Comment by sirpy** [comment]
@vldkhh @L03TJ3 @decentralauren 
i've noticed the following issues
- [ ] balances not updated after I do the bridge TX. (need to trigger a refresh)
- [ ] when I had no recent transfers the spinner was always loading.
- [ ] recent transfer doesn't look very good on mobile size (see image)
- [ ] no balance error text is not the best (see image)

<img width="452" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBN3Y1QlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--90f8118a9b79d8ee2955ecf16dc795ee7e56e5c6/image.png" alt="image.png" />

<img width="447" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNzM1QlE9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--385588fb3c31e3fadd6bda41b87d3f2358c588ba/image.png" alt="image.png" />

**Comment by vldkhh** [comment]
@sirpy so what is the status of this issues?

**Comment by decentralauren** [comment]
@sirpy good catches. I'm ok with it looking a little crunched on mobile and will create a separate ticket for that. However we should fix the error message before we push to prod as indeed it's confusing. This should say:

"Sorry, you don't have enough G$." 

Does the checkmark next to  these in your comment above mean that you've fixed them? Or that they still need to be fixed?

- balances not updated after I do the bridge TX. (need to trigger a refresh)
- when I had no recent transfers the spinner (in recent transactions) was always loading.

**Comment by sirpy** [comment]
@decentralauren yes they have been fixed.
@vldkhh it can be tested

**Comment by vldkhh** [comment]
@sirpy The microbridge on dev.gooddapp - made 2 tx from Celo to Fuse 100 000 G$ each. They are reflected as pending on the front but on Celoscan, they show as Successful. 
The Balance on Celo is showing properly (- 200 000 G$), but on the Fuse chain the balance is not updated. 

Also this point " balances not updated after I do the bridge TX. (need to trigger a refresh) " is not working properly.

@sirpy  Could you recheck on your side?

**Comment by sirpy** [comment]
@vldkhh please check again

**Comment by vldkhh** [comment]
verified on prod
